atlasdb-cassandra/src/main/java/org/apache/commons/pool2/impl/CommonsPoolGenericObjectPool.java

```diff
@@ -774,14 +777,31 @@ public void evict() throws Exception {
                     } catch (final NoSuchElementException nsee) {
                         // Object was borrowed in another thread
                         // Don't count this as an eviction test so reduce i;
+                        // CHANGELOG: Additional logging
+                        if (log.isDebugEnabled()) {
+                            log.debug("Not counting an eviction candidate, because we believe it was borrowed in "
```

Comment:
I suggest that we also log:
* `i`
* `m`
* some sort of UUID per eviction run. Then we can group logs by eviction run to confirm that it is indeed the same run that is looping indefinitely.

Reply:
Makes sense.